dds/DCPS/security/framework/SecurityConfig.h

```diff
@@ -111,6 +118,7 @@ class OpenDDS_Dcps_Export SecurityConfig : public DCPS::RcObject {
   CryptoKeyExchange_var key_exchange_plugin_;
   CryptoKeyFactory_var key_factory_plugin_;
   CryptoTransform_var transform_plugin_;
+  HandleRegistry* handle_registry_plugin_;
```

Comment:
Can you even do this? Isn't the security config across the whole application? What if you have two secure local participants talking to the same secure remote participant? Won't the handle lookups clash?

Reply:
Can, yes.  Should, no...for the exact reason you mentioned.  Interestingly enough, the existing approach to security "works" by using opaque handles so it is safe to share that state between participants.

I think what I'll do is create other class called `SecurityContext` to hold a participants security-related state that is 1) not explicitly handled in the SecurityConfig and 2) must be shared across module boundaries (the main one being security and transport).  This creates the following problem (which goes back to the "design" of the Transport framework):  who owns the SecurityContext?  Discovery?  The transport?  The transport framework?  Or does the SecurityConfig gain an additional map of participant to SecurityContext.  I think I'll go with the last one unless there is a strong objection.